.github/workflows/bump-version-and-publish.yml

```diff
@@ -0,0 +1,75 @@
+name: Bump version and publish release nuget
+
+on:
+  pull_request:
+    types: [ closed ]
+    branches: [ "master" ] 
+
+env:
+  NuGetDirectory: ${{ github.workspace }}/nuget
+
+jobs:
+  bump_version_and_add_tag:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.merge_commit_sha }}
+          fetch-depth: '0'
+      - name: Bump version and push tag
+        uses: anothrNick/github-tag-action@1.64.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          WITH_V: false
+          DEFAULT_BUMP: patch
+
+  build_test_and_pack:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-22.04
+    needs: [ bump_version_and_add_tag ]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Fetches entire history, so we can analyze commits since last tag
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: |
+            3.1.x
+            6.0.x
+      - name: Restore dependencies
+        run: dotnet restore
+      - name: Build
+        run: dotnet build --no-restore -c Release
+      - name: Test on .netcoreapp3.1
+        run: dotnet test -c Release -f netcoreapp3.1 --no-build --verbosity normal
+      - name: Test on .net6.0
+        run: dotnet test -c Release -f netcoreapp3.1 --no-build --verbosity normal
```

Comment:
Copy/paste error here and below. Probably, we need net6.0 here.

Reply:
Yes, thank you for noticing, fixed.

Reply:
@aensidhe if everything looks good, I think this PR is ready to be merged